### PR TITLE
Feature/API

### PR DESCRIPTION
**Pull request by yigitm**
In this PR:

- API endpoint to list all posts for a user is added,
- API endpoint to list all comments for a user's post is added,
- API endpoint to add a comment to a post is added,
- All API endpoints receive JSON & respond to JSON

Thank you for your time.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -8,5 +8,9 @@
     end
   end
 
+  get '/api/users/:id/posts', to: 'api#posts_index'
+  get '/api/users/:id/posts/:id', to: 'api#list_comments'
+  post '/api/users/:id/posts/:id', to: 'api#create_comment'
```
Comment (MrBrN197) [inline config/routes.rb]: - [ ] Hi, as part of the requirements for this project. you need to add two other routes.
- [ ] a Login API endpoint - POST : `/login`
- [ ] a Registration API endpoints - POST : `/register` or `/signup`

**Review by MrBrN197 — CHANGES_REQUESTED** [review]
## Hi, @yigitm & @Anjacodes  👋 

You've done a really nice job so far.
There are some issues that you still need to work on to go to the next project but you are almost there!
## To highlight 👏👏
- ✔ your API endpoints are functioning correctly

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (Anjacodes) [reply config/routes.rb]: Thank you for your feedback @MrBrN197, we have added a login and register endpoint now.

File: app/controllers/api/api_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      @new_comment = @user.comments.new(post_id: params[:post_id], text: params[:text])
+
+      if @new_comment.save
+        render json: @new_comment, status: :created
+      else
+        render json: @new_comment.errors, status: :unprocessable_entity
+      end
+    end
+
+  end
```
Comment (Emmyn5600) [inline app/controllers/api/api_controller.rb:27]: image

image

**Review by Emmyn5600 — APPROVED** [review]
Hi @yigitm and @Anjacodes 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! :tada: :clap: :clap: :100:

nodding-yes

## Optional suggestions

- [ ] N/A

##### _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

##### Cheers and Happy coding!:clap::clap::clap:
##### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/api/api_controller.rb (hunk 2 above)
Comment (Anjacodes) [reply app/controllers/api/api_controller.rb:27]: Thank you @MrBrN197 and [Emmyn5600](https://github.com/Emmyn5600). You just made our day!

**Pull request merged**